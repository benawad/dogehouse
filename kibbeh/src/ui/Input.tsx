/* eslint-disable no-unused-expressions */
import React, { forwardRef } from "react";

export interface InputProps extends React.ComponentPropsWithoutRef<"input"> {
  textarea?: boolean;
  rows?: number;
  error?: string;
  transparent?: boolean;
}

export const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ className, textarea, error, transparent, ...props }, ref) => {
    const bg = transparent ? `bg-transparent` : `bg-primary-700`;
    const ring = error ? `ring-1 ring-secondary` : "";
    const cn = `w-full py-2 px-4 rounded-8 text-primary-100 placeholder-primary-300 focus:outline-none ${bg} ${ring} ${className} `;

    return textarea ? (
      <textarea
        ref={ref as any}
        className={cn}
        data-testid="textarea"
        {...(props as any)}
      />
    ) : (
      <input ref={ref} className={cn} data-testid="input" {...props} />
    );
  }
);

Input.displayName = "Input";
