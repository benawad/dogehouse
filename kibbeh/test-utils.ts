import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
