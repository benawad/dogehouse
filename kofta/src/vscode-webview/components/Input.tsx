import React, { forwardRef } from "react";

export const Input = forwardRef<
  HTMLInputElement,
  React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > & { textarea?: boolean }
>(({ textarea, ...props }, ref) => {
  const cn = `w-full py-2 px-3 text-white bg-simple-gray-3c`;

  return textarea ? (
    <textarea ref={ref as any} className={cn} {...(props as any)} />
  ) : (
    <input ref={ref} className={cn} {...props} />
  );
});
