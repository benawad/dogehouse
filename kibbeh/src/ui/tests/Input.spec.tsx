import React from "react";

import { render } from "../../../test-utils";
import { Input, InputProps } from "../Input";

// jest.useFakeTimers();

// const friendOnline: FriendOnlineType = {
//   avatar: "",
//   isOnline: true,
//   username: "Some name",
//   activeRoom: {
//     name: "Some room",
//     link: "",
//   },
// };

// const showMoreAction = jest.fn();

// let defaultProps: InputProps = {
//   onlineFriendCount: 1,
//   onlineFriendList: [friendOnline],
//   showMoreAction,
// };

describe("Input", () => {
  it("should render and input correctly", () => {
    const { getByTestId } = render(<Input />);
    const component = getByTestId("input");

    expect(component).toBeVisible();
  });

  it("should render and input correctly", () => {
    const { getByTestId } = render(<Input textarea={true} />);
    const component = getByTestId("textarea");

    expect(component).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<Input />);
    const component = getByTestId("input");
    expect(component).toMatchSnapshot();
  });
});
