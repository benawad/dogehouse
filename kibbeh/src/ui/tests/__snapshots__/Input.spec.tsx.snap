// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  class="w-full py-2 px-4 rounded-8 text-primary-100 placeholder-primary-300 focus:outline-none bg-primary-700  undefined "
  data-testid="input"
/>
`;
